### PR TITLE
Add project icons

### PR DESCRIPTION
**Pull request by mazevedofs**
**Phabricator:** https://phabricator.wikimedia.org/T342938

### Notes
* This PR adds SwiftUI project icons to the components package

### Test steps
* Run the demo app, click on Project Icons
* Make sure the theming works

### CODE REVIEW

File: Demo/Demo/SampleViewController.swift [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import SwiftUI
+import UIKit
+import Components
+import WKData
+
+final public class SampleViewController: WKCanvasViewController {
```
Comment (staykids) [inline Demo/Demo/SampleViewController.swift]: 1. Looks like this accidentally got added to the root of the Demo project twice – can you remove this reference at the root?
svc

2. Could we rename this (and related `Sample` classes here) to something more descriptive of the feature? Maybe `ProjectIconSample...` or something of the sort?

File: Sources/Components/Components/Project Icons/WKProjectIconView.swift [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+public struct WKProjectIconView: View {
+
+    public var project: WKProject
+
+    public init(project: WKProject) {
+        self.project = project
+    }
+
+    @ObservedObject var appEnvironment = WKAppEnvironment.current
```
Comment (staykids) [inline Sources/Components/Components/Project Icons/WKProjectIconView.swift]: Minor: can we move this up next to the other property (`var project`)?

File: Sources/Components/Components/Project Icons/WKProjectIconView.swift [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+    public var body: some View {
+        switch project {
+        case .wikipedia(let wKLanguage):
+            let capitalizedText = wKLanguage.languageCode.uppercased()
```
Comment (staykids) [inline Sources/Components/Components/Project Icons/WKProjectIconView.swift]: To respect Locale, perhaps we should use `wKLanguage.languageCode.localizedUppercase` here instead.

File: Sources/Components/Components/Project Icons/WKProjectIconView.swift [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+                    .foregroundColor(Color(appEnvironment.theme.secondaryText))
+                    .padding([.leading, .trailing], 3)
+                    .padding([.top, .bottom], 4)
+
+            }
+            .overlay(RoundedRectangle(cornerRadius: 4)
+                .stroke(Color(appEnvironment.theme.secondaryText), lineWidth: 1)
+            )
+
+        case .wikidata:
```
Comment (staykids) [inline Sources/Components/Components/Project Icons/WKProjectIconView.swift:30]: I think when `WKProject` supports more project types, we should also match the same project source images currently supported in the main client app.

**Review by staykids — CHANGES_REQUESTED** [review]
Awesome – looking great! Some minor notes below.

File: Demo/Demo/SampleViewController.swift (hunk 1 above)
Comment (tonisevener) [reply Demo/Demo/SampleViewController.swift]: Done!

File: Sources/Components/Components/Project Icons/WKProjectIconView.swift (hunk 4 above)
Comment (tonisevener) [reply Sources/Components/Components/Project Icons/WKProjectIconView.swift:30]: @staykids Agreed - the double enum with `WikimediaProject` in the client app is really annoying. I'm happy to do this cleanup post-watchlist (we could also use some colors deduping) if you're okay with me kicking it down the road.

**Comment by tonisevener** [comment]
@staykids Ready for another look!

**Review by staykids — APPROVED** [review]
Nice work @mazevedofs and thanks for the assist @tonisevener!

**Pull request merged**